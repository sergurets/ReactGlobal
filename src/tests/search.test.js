import React from 'react';
import { shallow, mount } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import Search from '../components/search';
describe('Search', () => {
  it('should render correctly', () => {
    const output = shallow(
      <Search/>
    );
    expect(shallowToJson(output)).toMatchSnapshot();
  });
});

describe('Test Button component', () => {
    it('Test click event', () => {
      const mockCallBack = jest.fn();
      const button = shallow(<Search onClick={mockCallBack}/>);
      button.find('.mainSearchButton').simulate('click');
      expect(mockCallBack.mock.calls.length).toEqual(1);
    });
  });

 it("onChange param is the same value as the input value", () => {
    const input = mount(<Search/>).find('input');
    var inputDOM = input.getDOMNode(); 
    input.simulate('change', { target: { value: 'Changed' } })
    expect(inputDOM.value).toEqual('Changed');
    });

   
    