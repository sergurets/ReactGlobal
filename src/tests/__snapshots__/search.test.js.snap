// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Search should render correctly 1`] = `
<div
  className="search"
>
  <input
    id="mainSearch"
    onChange={[Function]}
    type="text"
    value=""
  />
  <div
    className="searchButtons"
  >
    <div
      className="searchFilter"
    >
      <span>
        SEARCH BY
      </span>
      <button
        className="active"
        onClick={[Function]}
      >
        TITLE
      </button>
      <button
        onClick={[Function]}
      >
        GENRE
      </button>
    </div>
    <button
      className="mainSearchButton"
      onClick={[Function]}
    >
      Search
    </button>
  </div>
</div>
`;
